nemo/collections/tts/models/magpietts.py

```diff
@@ -97,6 +98,34 @@ def __init__(self, cfg: DictConfig, trainer: 'Trainer' = None):
         if trainer is not None:
             self.world_size = trainer.num_nodes * trainer.num_devices
 
+        # load codec
+        codec_model = AudioCodecModel.restore_from(cfg.get('codecmodel_path'), strict=False)
+        # del codec discriminator to free memory
+        del codec_model.discriminator
+        self._codec_model = codec_model
+        self._codec_model.freeze()  #Lightning does requires_grad = False and self.eval()
+
+        # Reserve special tokens (appended at the end of the codebook after the codec tokens)
+        # (the actual index is this value plus the number of codec tokens - do not use the Enum directy)
+        class SpecialAudioToken(Enum):
+            AUDIO_BOS = 0
+            AUDIO_EOS = 1
+            AUDIO_CONTEXT_BOS = 2
+            AUDIO_CONTEXT_EOS = 3
+            MASK_TOKEN = 4
+            NUM_SPECIAL_TOKENS = 5 # update this if you add more special tokens
+
+        # Set up codebook configuration
+        # TODO @rfejgin: replace these to calls to an API at the codec (not quantizer) level once those are added
+        self.num_audio_codebooks = self._codec_model.vector_quantizer.num_groups
+        num_reserved_audio_tokens = cfg.get('num_reserved_audio_tokens', 8)
+        assert num_reserved_audio_tokens >= SpecialAudioToken.NUM_SPECIAL_TOKENS.value, "Are are not enough reserved entries for all special tokens"
+        self.num_audio_tokens_per_codebook = self._codec_model.vector_quantizer.codebook_size_per_group + num_reserved_audio_tokens
+        self.audio_bos_id = self.num_audio_tokens_per_codebook + SpecialAudioToken.AUDIO_BOS.value
+        self.audio_eos_id = self.num_audio_tokens_per_codebook + SpecialAudioToken.AUDIO_EOS.value
+        self.context_audio_bos_id = self.num_audio_tokens_per_codebook + SpecialAudioToken.AUDIO_CONTEXT_BOS.value
+        self.context_audio_eos_id = self.num_audio_tokens_per_codebook + SpecialAudioToken.AUDIO_CONTEXT_EOS.value
```

Comment:
self.num_audio_tokens_per_codebook is a misnomer. It is the total number of tokens in a codebook, maybe we should rename it to self.num_tokens_per_codebook or self.num_all_tokens_per_codebook or self.num_total_tokens_per_codebook.